src/app/host/[room]/page.tsx

```diff
@@ -1,46 +1,150 @@
 // HOST ROOM
 'use client'
 
+import Link from 'next/link'
 import { useEffect, useState } from 'react'
-import SocketIoInfo from '@/components/socketIoDevTools/SocketIoInfo'
-
 import { socketRoomEmitter } from '@/services/socket'
+import { useSocketListener } from '@/services/socket'
+import type { ListenerRes } from '@/services/socket'
+import HostControlButton from '@/components/HostControlButton'
 import HostSettingsButton from '@/components/HostSettingsButton'
-import HostControls from '@/components/HostControls'
 import RoomMainUi from '@/components/RoomMainUi'
+import RoomInfo from '@/components/RoomInfo'
 
-export default function HostRooms({ params }: { params: { room: string } }) {
-	const { room } = params
+export default function HostRoom({ params }: { params: { room: string } }) {
+	const [allUserPointData, setAllUserPointData] = useState<ListenerRes[]>([])
+	const storyPointRes = useSocketListener('story-points')
+	const joinRoomRes = useSocketListener('join-room')
 
-	console.log('%c>>> HOST ROOM roomId:', 'color: #5f0', room)
+	useEffect(() => {
```

Comment:
todo: This 2 `useEffect` can be replaced by the socket callback pattern we spoke in our call :)

Reply:
I don't believe it can. I have tried many different ways to use a callback pattern with these, and none work. Plus, we have the problem of it really not making any sense to do this. At least to me.

Reply:
Well done, yes, that's one of the reasons! In this case, useEffects adds extra re-render, and it split the data logic in different places. :)

Reply:
DONE